core/src/main/java/bio/terra/pearl/core/service/publishing/PortalDiffService.java

```diff
@@ -83,15 +88,43 @@ public PortalEnvironmentChange diffPortalEnvs(PortalEnvironment sourceEnv, Porta
             studyEnvChanges.add(studyEnvChange);
         }
 
+        List<ParticipantDashboardAlert> destAlerts = new ArrayList<>(destEnv.getParticipantDashboardAlerts());
+        List<ParticipantDashboardAlert> sourceAlerts = new ArrayList<>(sourceEnv.getParticipantDashboardAlerts());
+        List<ParticipantDashboardAlertChange> alertChangeLists = diffAlertLists(sourceAlerts, destAlerts);
+
         return new PortalEnvironmentChange(
                 siteContentRecord,
                 envConfigChanges,
                 preRegRecord,
                 notificationConfigChanges,
+                alertChangeLists,
                 studyEnvChanges
         );
     }
 
+    protected List<ParticipantDashboardAlertChange> diffAlertLists(
+            List<ParticipantDashboardAlert> sourceAlerts,
+            List<ParticipantDashboardAlert> destAlerts) throws Exception {
+        List<ParticipantDashboardAlert> unmatchedDestAlerts = new ArrayList<>(destAlerts);
```

Comment:
Might be more efficient to represent `unmatchedDestAlerts` as a map of trigger to alert.